src/Components/Server/src/BlazorPack/BlazorPackHubProtocolWorker.cs

```diff
@@ -34,6 +35,20 @@ protected override object DeserializeObject(ref MessagePackReader reader, Type t
                 {
                     return reader.ReadSingle();
                 }
+                else if (type == typeof(byte[]))
+                {
+                    var bytes = reader.ReadBytes();
+                    if (!bytes.HasValue)
+                    {
+                        return null;
+                    }
+                    else if (bytes.Value.Length == 0)
+                    {
+                        return Array.Empty<byte>();
+                    }
+
+                    return bytes.Value.ToArray();
```

Comment:
Still doing `ToArray` in place of passing through the `ReadOnlySequence<byte>`. 

Concerns with using the `ReadOnlySequence` relate to the fact that it doesn't implicitly cast to `byte[]` so when we're examining method parameters we have a type mistmatch.

Additionally, as Javier mentioned in the other PR we need to verify the lifecycle/ownership of the `ReadOnlySequence`.

Reply:
> Concerns with using the ReadOnlySequence relate to the fact that it doesn't implicitly cast to byte[] so when we're examining method parameters we have a type mistmatch.

I think you'd need to change the receiving code (`ComponentHub.SupplyByteArray`) to accept a `ReadOnlySequence<byte>` instead of a `byte[]`, so there wouldn't be a type mismatch.

Agreed on verifying the ownership of the buffer before doing that though.

Reply:
Yes, lets keep it as is for the time being and then we can do a quick check and change afterwards.